packages/mdc-dialog/index.js

```diff
@@ -171,7 +172,10 @@ class MDCDialog extends MDCComponent {
       releaseFocus: () => this.focusTrap_.deactivate(),
       isContentScrollable: () => !!this.content_ && util.isScrollable(/** @type {!Element} */ (this.content_)),
       areButtonsStacked: () => util.areTopsMisaligned(this.buttons_),
-      getActionFromEvent: (event) => event.target.getAttribute(strings.ACTION_ATTRIBUTE),
+      getActionFromEvent: (event) => {
+        const element = closest(event.target, `[${strings.ACTION_ATTRIBUTE}]`);
```

Comment:
Not in the scope of this PR, But, just like MDC List custom event proposal (#3481), button / icon button can also have a custom event that encapsulates this logic.

Reply:
That's making a strong assumption that those are the only things to ever report an action. List items will also report actions, and I'd rather remain unopinionated on what types of things people assign actions to (e.g. links?)

Previously MDC Dialog has received complaints because it was way too strongly opinionated about actions. The way it's reimplemented makes it pretty flexible.

Reply:
It is up to consumer whether to use the custom event or go with traditional approach. The benefit of proxying native click event I think is that not every wrapper has to walk through the DOM to find the closest. But this looks ok for now.